src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -343,7 +343,7 @@ def execute(self, args):
                 logger.warning(d.message)
 
             try:
-                result = cmd(params)
+                result = cmd(self._copy_params(params))
```

Comment:
Rather than this copy, why can't we simply not pop the argument values in the generic update logic?

Reply:
So the reason I copied the arguments was to make `--ids` more robust so that if a command / feature is added in the future that inadvertently mutates an argument the command would still be successful.

Reply:
This imposes an unnecessary and wasteful overhead on every command, including ones that don't use multiple IDs. The real problem is a bug in the logic of generic update that assumes a single invocation. Thus, the fix should be to correct the logic bug, not impose a workaround.

Reply:
That's fair. I will correct the logic bug in the logic of generic update